File: Practice/28/c++/28/28.cpp
#include <iostream>
#include <map>

using namespace std;
typedef map<int, int> new_Map;

void print_factorization(const new_Map& factors) {
	int len_factors = factors.size();
	int i = 0;
	for (auto item : factors) {
		i++;
		cout << item.first;
		if (item.second > 1) {
			cout << "^" << item.second;
		}
		if (i != len_factors) {
			cout << " * ";
		}
	}
}

void factorization(unsigned int num)
{
	new_Map factors;
	while (num > 1){
		for (int i = 2; i <= num; i++) {
			if (num % i == 0) {
				num /= i;
				factors[i] = factors[i] + 1;
				break;
			}
		}
	}
	print_factorization(factors);
}

int main()
{
	setlocale(LC_ALL, "ru");
	int num;
	cout << "Введите число: ";
	cin >> num;
	factorization(num);
}

File: Practice/28/python/28.py
def print_factorization(factors):
    number_of_keys = len(factors.keys())
    i = 0
    for k, v in factors.items():
        i += 1
        print(k, end='')
        if v > 1:
            print(f'^{v}', end='')
        if i != number_of_keys:
            print(' * ', end='')


def factorization(n):
    factors = {}
    while n > 1:
        for i in range(2, n + 1):
            if n % i == 0:
                n //= i
                factors[i] = factors.get(i, 0) + 1
                break
    print_factorization(factors)


num = int(input('Введите число: '))
factorization(num)
